src/extension/background-script/actions/nostr/getPublicKey.ts

```diff
@@ -0,0 +1,22 @@
+import * as secp256k1 from "@noble/secp256k1";
+import { decryptData } from "~/common/lib/crypto";
+import state from "~/extension/background-script/state";
+
+const getPublicKey = async () => {
+  const encryptedPrivateKey = state.getState().nostrPrivateKey;
+  if (!encryptedPrivateKey) {
+    return;
+  }
+  const decryptedPrivateKey = decryptData(
+    encryptedPrivateKey,
+    state.getState().password as string
+  );
+  const buffer = Buffer.from(decryptedPrivateKey, "hex");
+  const publicKey = Buffer.from(
```

Comment:
I think instead of the `Buffer` you can use `secp.utils.bytesToHex` 
(Buffer is some polyfill and thus I'd avoid it as much as possible?)

Reply:
good point, thanks for the hint